wikimedia/wikipedia-ios
# Notifications - add swipe actions from cells

**Phabricator:**
https://phabricator.wikimedia.org/T288687 (data and routing portion, mark as read to come in a followup PR)

### Notes
This PR establishes the swipe action label and routing urls for each notification. It builds on the swipe gesture recognizer from https://github.com/wikimedia/wikipedia-ios/pull/4063 and the url generation helper methods in https://github.com/wikimedia/wikipedia-ios/pull/4064, so do not merge until those are in place. For now we are using a standard action sheet / popover to display these options - it will be replaced with a custom sheet in the future.

### Test Steps
1. Run on the main/production scheme with an account that has lots of notifications. It's helpful if you're able to find an account with notifications on Commons, Wikidata, and multiple languages. Be sure to add those other languages as an app language in Settings before going to Notification Center. I can share my testing credentials if needed.
2. Browse through notification center via the Explore bell icon. Swipe right on each cell and confirm the action sheet labels look right. Tap one of the custom options in the action sheet (i.e. not mark as read or notification settings). Confirm the user routes to the correct place.

**Optional**
I have a custom payload of notifications I've gathered to mimic some harder-to-reach notifications. If interested you can proxy and map the https://en.wikipedia.org/w/api.php?action=query&format=json&formatversion=2&meta=notifications&notfilter=read%7C!read&notformat=model&notlimit=max&notwikis=enwiki call to this before navigating to Notifications Center. In general though, we shouldn't get too in depth on this testing and instead lean on QA. Feel free to reach out to me for this file.
